scene/2d/line_builder.cpp

```diff
@@ -124,32 +76,42 @@ void LineBuilder::build() {
 	float current_distance0 = 0.f;
 	float current_distance1 = 0.f;
 	float total_distance = 0.f;
-	float width_factor = 1.f;
+
 	_interpolate_color = gradient != nullptr;
 	bool retrieve_curve = curve != nullptr;
+	float width_factor = 1.f;
+	if (retrieve_curve) {
+		width_factor = curve->sample_baked(0.f);
+		modified_hw = hw * width_factor;
+	}
+
 	bool distance_required = _interpolate_color ||
 			retrieve_curve ||
 			texture_mode == Line2D::LINE_TEXTURE_TILE ||
 			texture_mode == Line2D::LINE_TEXTURE_STRETCH;
 	if (distance_required) {
-		total_distance = calculate_total_distance(points);
-		//Adjust totalDistance.
-		// The line's outer length will be a little higher due to begin and end caps
+		// Calculate the total distance.
+		for (int i = 1; i < point_count; ++i) {
+			total_distance += points[i].distance_to(points[i - 1]);
+		}
+		if (wrap_around) {
+			total_distance += points[point_count - 1].distance_to(points[0]);
+		}
+
+		// Adjust the total distance.
+		// The line's outer length will be a little higher due to begin and end caps.
 		if (begin_cap_mode == Line2D::LINE_CAP_BOX || begin_cap_mode == Line2D::LINE_CAP_ROUND) {
-			if (retrieve_curve) {
-				total_distance += width * curve->sample_baked(0.f) * 0.5f;
-			} else {
-				total_distance += width * 0.5f;
-			}
+			total_distance += modified_hw;
 		}
 		if (end_cap_mode == Line2D::LINE_CAP_BOX || end_cap_mode == Line2D::LINE_CAP_ROUND) {
 			if (retrieve_curve) {
-				total_distance += width * curve->sample_baked(1.f) * 0.5f;
+				total_distance += hw * curve->sample_baked(1.f);
 			} else {
-				total_distance += width * 0.5f;
+				total_distance += hw;
 			}
 		}
```

Comment:
This shouldn't be done when wrapping around. Artifact in action:

||||
|-|-|-|
|![t65vK67EMQ](https://github.com/godotengine/godot/assets/9283098/58f93c42-08ef-4c2c-8af1-a53887dcebdb)|![BwfHCVdKSh](https://github.com/godotengine/godot/assets/9283098/d7da67a0-79bb-4843-b27d-997448e1228f)|![Jgz8w2ORiW](https://github.com/godotengine/godot/assets/9283098/391e26af-0698-4dd9-b03a-ff0adc07f59c)|

These ifs can be moved into an `else` block for the preceding `if (wrap_around)` check.